rfcs/0000-execute-builds-in-standard-containers.md

```diff
@@ -0,0 +1,31 @@
+**Problem:**
+
+Currently, kpack executes each stage of the [CNB Lifecycle](https://buildpacks.io/docs/concepts/components/lifecycle/) inside of individual [init containers]((https://kubernetes.io/docs/concepts/workloads/pods/init-containers/)). This has two major benefits:
+
+1. 	Different stages of the lifecycle are isolated from one another allowing the credentials required to push to the registry to be only mounted in the containers that need them, preventing a rogue buildpack from reading the secrets.
+2. We can very easily define the order of the steps occuring as well as waiting for the previous step to complete before continuining on. Additionally, failures in one init container cause the entire build to fail without additional logic on our part
+
+As kpack has grown, we have seen a need to support service meshes such as [Istio](https://istio.io/latest/about/service-mesh/) which do not work when you have init containers that need to reach out to the network. This is a [well documented](https://istio.io/latest/docs/setup/additional-setup/cni/#compatibility-with-application-init-containers) technical limitation inside Istio. The reason this occurs is because Istio adds a sidecar container to the kpack build pod which runs [envoy proxy](https://www.envoyproxy.io/) and sets up traffic from the build containers to be routed through the proxy. Due to the fact that init containers run before any normal container can start, kpack builds are unable to reach the network because they are attempting to communicate through the envoy proxy container that has not and will not start until the build completes.
+
+To get around this issue, kpack currently applies an annotation (`sidecar.istio.io/inject=false`) to all build pods to disable istio sidecar injection. This presents a problem where users might want to take advantage of the various features of Istio, but cannot because kpack is bypassing it.
+
+**Outcome:**
+
+kpack builds can execute successfully when an istio sidecar is injected into the pod
+
+**Actions to take:**
+
+**Complexity:**
+
+**Prior Art:**
+Tekton: `https://github.com/tektoncd/pipeline/tree/main/cmd/entrypoint#waiting-for-sidecars`
+kpack spike: `https://github.com/pivotal/kpack/tree/execute-builds-in-regular-containers`
+
+**Alternatives:**
+
+* We can build the Tekton entrypoint code as an image to use in kpack
+* We could execute all build steps in one standard container similar to pack build with the `--trust-builder` flag 
```

Comment:
How would this work with the prepare step?

Reply:
My thinking was that we would create our own binary that runs the prepare step and then execs the lifecycle steps inside one container. I don't intend for this to be a really viable solution but I was just trying to call out that it is the other end of the spectrum of solutions